features/step_definitions/git_steps.rb

```diff
@@ -1,48 +1,75 @@
-Given 'a mocked git configuration' do
-  steps %(
-    Given a mocked home directory
-    And I run `git config --global user.name Test`
-    And I run `git config --global user.email test@example.com`
-  )
-end
+require_relative '../../spec/helpers/faker/puppet_module_remote_repo'
 
-Given 'a remote module repository' do
+Given 'a basic setup with a puppet module {string} from {string}' do |name, namespace|
   steps %(
-    Given a directory named "sources"
-    And I run `git clone https://github.com/maestrodev/puppet-test sources/puppet-test`
+    Given a mocked git configuration
+    And a puppet module "#{name}" from "#{namespace}"
     And a file named "managed_modules.yml" with:
       """
       ---
-        - puppet-test
+        - #{name}
       """
+    And a file named "modulesync.yml" with:
+      """
+      ---
+      namespace: #{namespace}
+      """
+    And a git_base option appended to "modulesync.yml" for local tests
   )
-  write_file('modulesync.yml', <<~CONFIG)
-    ---
-    namespace: sources
-    git_base: file://#{expand_path('.')}/
-    CONFIG
 end
 
-Given Regexp.new(/a remote module repository with "(.+?)" as the default branch/) do |branch|
+Given 'a mocked git configuration' do
   steps %(
-    Given a directory named "sources"
-    And I run `git clone --mirror https://github.com/maestrodev/puppet-test sources/puppet-test`
-    And a file named "managed_modules.yml" with:
-      """
-      ---
-        - puppet-test
-      """
+    Given a mocked home directory
+    And I run `git config --global user.name Aruba`
+    And I run `git config --global user.email aruba@example.com`
   )
-  write_file('modulesync.yml', <<~CONFIG)
-    ---
-    namespace: sources
-    git_base: file://#{expand_path('.')}/
-    CONFIG
-
-  cd('sources/puppet-test') do
-    steps %(
-      And I run `git branch -M master #{branch}`
-      And I run `git symbolic-ref HEAD refs/heads/#{branch}`
-    )
-  end
+end
+
+Given 'a puppet module {string} from {string}' do |name, namespace|
+  pmrr = Faker::PuppetModuleRemoteRepo.new(name, namespace)
+  pmrr.populate
+end
+
+Given 'a git_base option appended to "modulesync.yml" for local tests' do
+  File.write "#{Aruba.config.working_directory}/modulesync.yml", "\ngit_base: #{Faker::PuppetModuleRemoteRepo.git_base}", mode: 'a'
+end
+
+Given 'the puppet module {string} from {string} is read-only' do |name, namespace|
+  pmrr = Faker::PuppetModuleRemoteRepo.new(name, namespace)
+  pmrr.read_only = true
+end
+
+Then 'the puppet module {string} from {string} have no commit between {string} and {string}' do |name, namespace, commit1, commit2|
```

Comment:
```suggestion
Then 'the puppet module {string} from {string} has no commits between {string} and {string}' do |name, namespace, commit1, commit2|
```

Reply:
Will use `should have` as its an expectation, good catch!